azure/services/async/async.go

```diff
@@ -82,9 +110,21 @@ func CreateResource(ctx context.Context, scope FutureScope, client Creator, spec
 		return processOngoingOperation(ctx, scope, client, resourceName, serviceName)
 	}
 
-	// No long running operation is active, so create the resource.
+	// No long running operation is active, so get the existing resource.
```

Comment:
IMO the previous comment still makes sense, we're still creating the resource. Getting the existing resource to check if anything needs to be updated before doing a create is an implementation detail

maybe something like
```suggestion
	// No long-running operation is active, so get or create the resource.
```
makes sense?